File: 933-increasing-order-search-tree/increasing-order-search-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    priority_queue<int, vector<int>, greater<int>> pq;
    void bc(TreeNode* root){
        if(!root) return;
        pq.push(root->val);
        bc(root->left);
        bc(root->right);
    }
    TreeNode* increasingBST(TreeNode* root) {
        bc(root);
        TreeNode* t = new TreeNode(pq.top());
        pq.pop();
        TreeNode* t1 = t;
        while(!pq.empty()){
            t1->right = new TreeNode(pq.top());
            t1 = t1->right;
            pq.pop();
        }
        return t;
    }
};

File: 933-increasing-order-search-tree/increasing-order-search-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */



var increasingBST = function(root) {
    if(root.right == undefined && root.left == undefined) return root;
    let arr = [];
    function getAll(c){
        if(c == undefined) return;
        arr.push(c.val);

        getAll(c.left);
        getAll(c.right);
    }

    getAll(root);
    arr.sort((a, b) => a - b)
    console.log(arr)
    
    
    let tmp = new TreeNode(arr[0], undefined, new TreeNode());
    let tmp2 = tmp;
    for(let i = 1; i<arr.length;i++){
        tmp2.right = new TreeNode(arr[i], undefined, undefined);
        tmp2 = tmp2.right;
    }

    return tmp;
};